packages/cozy-client/src/CozyClient.js

```diff
@@ -1094,6 +1099,36 @@ client.query(Q('io.cozy.bills'))`)
     return withIncluded
   }
 
+  /**
+   * Save the document or array of documents into the persisted storage (if any)
+   *
+   * @private
+   * @param {CozyClientDocument | Array<CozyClientDocument>} data - Document or array of documents to be saved
+   * @returns {Promise<void>}
+   */
+  async persistVirtualDocuments(data) {
+    if (!Array.isArray(data)) {
+      await this.persistVirtualDocument(data)
+    } else {
+      for (const document of data) {
```

Comment:
Yes, but I don't think this will cost that much as this is a pure JS loop that check objects properties. Also I don't have better solution, so I propose to keep it that way for now but to be attentive about performances

Reply:
> but to be attentive about performances

I'm pretty sure we won't detect this overhead. I guess there would be some way to improve, but let's keep it simple for now